src/main/java/com/amazonaws/services/kinesis/multilang/MultiLangDaemon.java

```diff
@@ -140,11 +143,26 @@ public static void main(String[] args) {
         ExecutorService executorService = config.getExecutorService();
 
         // Daemon
-        MultiLangDaemon daemon = new MultiLangDaemon(
+        final MultiLangDaemon daemon = new MultiLangDaemon(
                 config.getKinesisClientLibConfiguration(),
                 config.getRecordProcessorFactory(),
                 executorService);
 
+        Runtime.getRuntime().addShutdownHook(new Thread() {
+            @Override
+            public void run()
+            {
+                LOG.info("Process terminanted, will initiate shutdown.");
+                try {
+                    Future<Void> fut = daemon.worker.requestShutdown();
```

Comment:
You need to ensure that when `requestShutdown ` is invoked. There is no process records invoked. Need a lock may be ?!